pages/docs/reference/whatsnew14.md

```diff
@@ -574,3 +574,363 @@ Complete the initial configuration, and when you add a new dependency to `cocoap
+    
+    fun main() {
+    //sampleStart
+        val order = listOf<OrderItem>(
+            OrderItem("Cake", price = 10.0, count = 1),
+            OrderItem("Coffee", price = 2.5, count = 3),
+            OrderItem("Tea", price = 1.5, count = 2))
+    
+        val total = order.sumOf { it.price * it.count} // Double
+        val count = order.sumOf { it.count } // Int
+    //sampleEnd
+        println("You've ordered $count items that cost $total in total")
+    }
+    ```
+    </div>
+
+* `min()` and `max()` functions have been renamed to `minOrNull()` and `maxOrNull()` to comply with the naming
+  convention used across the Kotlin collections API: `*OrNull` suffix in the function name means that it returns `null`
+  if the receiver collection is empty. 
+    New `minOf()` and `maxOf()` extension functions return the minimum and the maximum value of the given selector function on the collection items.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    data class OrderItem(val name: String, val price: Double, val count: Int)
+    
+    fun main() {
+    //sampleStart
+        val order = listOf<OrderItem>(
+            OrderItem("Cake", price = 10.0, count = 1),
+            OrderItem("Coffee", price = 2.5, count = 3),
+            OrderItem("Tea", price = 1.5, count = 2))
+        val highestPrice = order.maxOf { it.price }
+    //sampleEnd
+        println("The most expensive item in the order costs $highestPrice")
+    }
+    ```
+    </div>
+
+    There are also `minOfWith()` and `maxOfWith()`, which take a `Comparator` as an argument, and `*OrNull()` versions
+of all four functions that return `null` on empty collections.
+
+* New overloads for `flatMap` and `flatMapTo` let you use transformations with return types that don’t match the receiver type, namely:
+    * transformations to `Sequence` on `Iterable`, `Array`, and `Map`
+    * transformations to `Iterable` on `Sequence`
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val list = listOf("kot", "lin")
+        val lettersList = list.flatMap { it.asSequence() }
+        val lettersSeq = list.asSequence().flatMap { it.toList() }    
+    //sampleEnd
+        println(lettersList)
+        println(lettersSeq.toList())
+    }
+    ```
+    </div>
+
+* `flatMapIndexed()` counterpart for `flatMap()`.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+        val list = listOf("hello", "kot", "lin", "world")
+        val kotlin = list.flatMapIndexed { index, item ->
+            if (index in 1..2) item.toList() else emptyList() 
+        }
+    //sampleEnd
+        println(kotlin)
+    }
+    ```
+    </div>
+
+We also add the `ArrayDeque` class - an implementation of double-ended queue.
```

Comment:
We also added new extension functions `MutableList.removeFirst() / removeFirstOrNull() / removeLast() / removeLastOrNull()` for mutable lists.